### PR TITLE
Remove deprecated IFluidSerializer from core-interfaces

### LINKED ISSUE #8682: Delete IFluidSerializer from core-interfaces
IFluidSerializer is deprecated as of v0.55 (it was moved to shared-object-base). It can be removed in a few versions and after IFluidObject references to it are removed.

### PR DESCRIPTION
**Pull request by scarlettjlee**
This was deprecated in 0.55 and can be removed now. Also remove it from IFluidObject which was deprecated in 0.52.

Closes #8682.

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -14,6 +14,19 @@ There are a few steps you can take to write a good change note and avoid needing
 - Provide guidance on how the change should be consumed if applicable, such as by specifying replacement APIs.
 - Consider providing code examples as part of guidance for non-trivial changes.
 
+# 1.0.0
```
Comment (heliocliu) [inline BREAKING.md:17]: 🥳🥳

**Review by heliocliu — APPROVED** [review]

**Review by vladsud — APPROVED** [review]

File: common/lib/core-interfaces/package.json [hunk 2]
```diff
@@ -77,6 +77,22 @@
         "RemovedInterfaceDeclaration_IProvideFluidConfiguration": {
           "backCompat": false,
           "forwardCompat": false
+        },
```
Comment (anthony-murphy) [inline common/lib/core-interfaces/package.json:80]: The version of this package needs to be updated. as right now it is the release version, it should be bumped to the next version, @heliocliu

Comment (heliocliu) [reply common/lib/core-interfaces/package.json:80]: working on it :d

Comment (anthony-murphy) [reply common/lib/core-interfaces/package.json:80]: in terms of process, we probably want bump next immediately after we merge to main. My main worry is pre-release packages coming from the different branches as people make and consume changes across the repo

Comment (heliocliu) [reply common/lib/core-interfaces/package.json:80]: that is how things are written in the instructions, but that didn't happen this time due to unexpected stuff with versions that was messing up the tooling plus some test failures that were messing up builds for the prerelease pipelines 🤕

**Pull request merged**